File: JewelsAndStones.java
//https://leetcode-cn.com/problems/jewels-and-stones/description/
class Solution {
      public int numJewelsInStones(String J, String S) {
        if (J == null || S == null)
            return 0;
        int count = 0;
        for (char s : S.toCharArray()) {
            for (char c : J.toCharArray()) {
                if (s == c)
                    count++;
            }
        }
        return count;

    }
}

File: JewelsAndStones.go
func numJewelsInStones(jewels string, stones string) int {
	res := 0
	jewelsSet := map[byte]bool{}

	for i := range jewels {
		jewelsSet[jewels[i]] = true
	}

	for i := range stones {
		if jewelsSet[stones[i]] {
			res++
		}
	}
	return res
}
